src/core/thread/topology.hpp

```diff
@@ -589,6 +595,57 @@ class Neighbor : public InstanceLocatorInit
     void SetTimeSyncEnabled(bool aEnabled) { mTimeSyncEnabled = aEnabled; }
 #endif
 
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+    /**
+     * This method aggregates the Link Metrics data into all the series that is running for this neighbor.
+     *
+     * If a series wants to account frames of @p aFrameType, it would add count by 1 and aggregrate @p aLqi and
+     * @p aRss into its averagers.
+     *
+     * @param[in] aSeriesId     Series ID for Link Probe. Should be `0` if this method is not called by Link Probe.
+     * @param[in] aFrameType    Type of the frame that carries Link Metrics data.
+     * @param[in] aLqi          The LQI value.
+     * @param[in] aRss          The Rss value.
+     *
+     */
+    void AggregateLinkMetrics(uint8_t aSeriesId, uint8_t aFrameType, uint8_t aLqi, int8_t aRss);
+
+    /**
+     * This method adds a new LinkMetricsSeriesInfo to the neighbor's list.
+     *
+     * @param[in]    A reference to the new LinkMetricsSeriesInfo.
+     *
+     */
+    void AddForwardTrackingSeriesInfo(LinkMetricsSeriesInfo &aLinkMetricsSeriesInfo);
+
+    /**
+     * This method finds a specific LinkMetricsSeriesInfo by Series ID.
+     *
+     * @param[in] aSeriesId    A reference to the Series ID.
+     *
+     * @returns The pointer to the LinkMetricsSeriesInfo. `nullptr` if not found.
+     *
+     */
+    LinkMetricsSeriesInfo *GetForwardTrackingSeriesInfo(const uint8_t &aSeriesId);
+
+    /**
+     * This method removes a specific LinkMetricsSeriesInfo by Series ID.
+     *
+     * @param[in] aSeriesId    A reference to the Series ID.
+     *
+     * @returns The pointer to the LinkMetricsSeriesInfo. `nullptr` if not found.
+     *
+     */
+    LinkMetricsSeriesInfo *RemoveForwardTrackingSeriesInfo(const uint8_t &aSeriesId);
```

Comment:
```suggestion
    LinkMetricsSeriesInfo *RemoveForwardTrackingSeriesInfo(const uint8_t aSeriesId);
```

Reply:
There is a reason for using reference. The underlying template method `RemoveMatching` (linked_list.hpp) requires to use const reference.